releasenotes/notes/fix-parameter-hash-d22c270090ffc80e.yaml

```diff
@@ -0,0 +1,13 @@
+---
+features:
+  - |
+    :class:`.Parameter` now has an advanced-usage keyword argument ``uuid`` in its constructor,
+    which can be used to make the :class:`.Parameter` compare equal to another of the same name.
+    This should not typically be used by users, and is most useful for custom serialisation and
+    deserialisation.
+fixes:
+  - |
+    The hash of a :class:`.Parameter` is now equal to the hashes of any
+    :class:`.ParameterExpression` that it compares equal to.  Previously the hashes were different,
+    which would cause spurious additional entries in hashmaps when :class:`.Parameter` and
+    :class:`.ParameterExpression` values were mixed in the same map.
```

Comment:
Maybe it should say "which violates Python's data model?"

Reply:
Done in 44711b4c98.